src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -275,68 +276,48 @@ public Task<ResultSetSubset> GetSubset(int batchIndex, int resultSetIndex, int s
                 return;
             }
 
-            // Open up a connection for querying the database
-            string connectionString = ConnectionService.BuildConnectionString(editorConnection.ConnectionDetails);
-            // TODO: Don't create a new connection every time, see TFS #834978
-            using (DbConnection conn = editorConnection.Factory.CreateSqlConnection(connectionString))
+            // Locate and setup the connection
```

Comment:
I think the changes to the query execution service and the connection service could have happened in separate pull requests and made the big changes easier to deal with. Buuuuut, here we are. File that away for the future 😃